sky/shell/BUILD.gn

```diff
@@ -220,6 +220,35 @@ if (is_android) {
       ":sky_shell",
     ]
   }
+
+  # The shell_lib target is used by the 'flutter apk' tool from the flutter
+  # repository.
+  standalone_lib = "$target_gen_dir/standalone_lib.dex.jar"
+  android_standalone_library("standalone_lib") {
+    dex_path = standalone_lib
+    deps = [
+      ":java",
+    ]
+  }
+
+  dex("shell_lib") {
+    deps = [
+      ":standalone_lib",
+      ":shell_apk",
+    ]
+    sources = [
+      standalone_lib,
+      "$target_gen_dir/shell_apk/shell_apk.jar",
+    ]
+    output = "$target_gen_dir/shell_lib.dex.jar"
+  }
+
+  group("shell") {
+    deps = [
+      ":shell_apk",
+      ":shell_lib",
+    ]
+  }
```

Comment:
We should restructure the build system to better match what we're doing.  Specifically, we should create the dylib for the C++ and the jar for the Java and then have a step that combines them into the APK.

It's not clear whether we should distribute a jar or a dex.jar.  What's the tradeoff between the two?